client/core/market.go

```diff
@@ -46,6 +46,18 @@ func (spotMarket SpotMarket) PriceFromChainFormat(chainValue cosmtypes.Dec) deci
 	return decimal.RequireFromString(chainValue.String()).Mul(decimal.New(1, decimals))
 }
 
+func (spotMarket SpotMarket) QuantityFromExtendedChainFormat(chainValue cosmtypes.Dec) decimal.Decimal {
+	return spotMarket.fromExtendedChainformat(spotMarket.QuantityFromChainFormat(chainValue))
+}
+
+func (spotMarket SpotMarket) PriceFromExtendedChainFormat(chainValue cosmtypes.Dec) decimal.Decimal {
+	return spotMarket.fromExtendedChainformat(spotMarket.PriceFromChainFormat(chainValue))
+}
+
+func (spotMarket SpotMarket) fromExtendedChainformat(chainValue decimal.Decimal) decimal.Decimal {
+	return chainValue.Div(decimal.New(1, AdditionalChainFormatDecimals))
```

Comment:
Better to do this in utils file, this one doesn't depend on `spotMarket` or `derivativeMarket`

```
fromExtendedChainformat(chainValue decimal.Decimal) decimal.Decimal
```

But why chainValue has decimal.Decimal type?

Reply:
Ok, will move it (unfortunately moving it to utils makes it less OOP, something that I don't really like)

Regarding the chainValue, it was not originally decimal.Decimal. But the function that calls this one first translates the value according to market tokens decimals, making it a decimal.Decimal

Reply:
> unfortunately moving it to utils makes it less OOP, something that I don't really like

Yeah it's a tool and we need flexibility to choose which to use

> Regarding the chainValue, it was not originally decimal.Decimal. But the function that calls this one first translates the value according to market tokens decimals, making it a decimal.Decimal

Oh I thought using sdk.Dec is enough when chain streamed the value